mne/io/ctf/tests/test_ctf.py

```diff
@@ -216,6 +218,42 @@ def test_read_ctf():
                   op.join(ctf_dir, ctf_fname_continuous), 'foo')
 
 
+@testing.requires_testing_data
+def test_rawctf_clean_names():
+    """Test RawCTF _clean_names method"""
+    # read test data
+    raw = read_raw_ctf(op.join(ctf_dir, ctf_fname_catch))
+    raw_cleaned = read_raw_ctf(op.join(ctf_dir, ctf_fname_catch),
+                               clean_names=True)
+    test_channel_names = _clean_names(raw.ch_names)
+    test_info_comps = copy.deepcopy(raw.info['comps'])
+
+    # channel names should not be cleaned by default
+    assert not array_equal(raw.ch_names, test_channel_names)
+
+    chs_ch_names = []
+    for ch in raw.info['chs']:
+        chs_ch_names.append(ch['ch_name'])
+
+    assert not array_equal(chs_ch_names, test_channel_names)
+
+    for test_comp, comp in zip(test_info_comps, raw.info['comps']):
+        for key in ('row_names', 'col_names'):
+            assert not array_equal(_clean_names(test_comp['data'][key]),
+                                   comp['data'][key])
+
+    # channel names should be cleaned if clean_names=True
+    assert array_equal(raw_cleaned.ch_names, test_channel_names)
+
+    for ch, test_ch_name in zip(raw_cleaned.info['chs'], test_channel_names):
+        assert ch['ch_name'] == test_ch_name
+
+    for test_comp, comp in zip(test_info_comps, raw_cleaned.info['comps']):
+        for key in ('row_names', 'col_names'):
+            assert array_equal(_clean_names(test_comp['data'][key]),
+                               comp['data'][key])
+
```

Comment:
The general rule now that we use `pytest` is use NumPy `numpy.testing.assert_allclose` or `numpy.testing.assert_array_equal` when dealing with NumPy arrays, and plain `assert ` otherwise (e.g., `assert a == 1.`, `assert b is not None`, or `assert c == [1, 3, 'foo']`). `pytest` does a nice job with plain `assert` statements for builtin datatypes.

Reply:
It looks like these are really list of str (builtin types) so I'd do `assert raw_cleaned.ch_names != test_channel_names`